### PR TITLE
Make adapter to notify user about changes

### PR DESCRIPTION
**Pull request by ZeRoGerc**
# Upload and deletion of posts implemented
 - Could be some optimizations in next requests

### CODE REVIEW

File: app/src/main/java/ly/loud/loudly/ui/adapters/FeedAdapter.java [hunk 1]
```diff
@@ -47,16 +55,42 @@ public int getItemCount() {
...
 
+    public void updateDeleted(@NonNull SolidList<PlainPost> newPosts) {
+        ArrayList<Integer> deletedPositions = new ArrayList<>();
+        ArrayList<PlainPost> updated = new ArrayList<>(posts.size());
+        int pointerNew = 0;
+        for (int i = 0, size = posts.size(); i < size; i++) {
+            PlainPost current = posts.get(i);
+            while (pointerNew < newPosts.size() && newPosts.get(pointerNew).getDate() < current.getDate()) {
+
+            }
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/ui/adapters/FeedAdapter.java]: Empty cycle's body?

File: app/src/main/java/ly/loud/loudly/ui/adapters/holders/ViewHolderPost.java [hunk 2]
```diff
@@ -91,6 +93,10 @@ public ViewHolderPost(@NonNull LayoutInflater inflater, @NonNull ViewGroup paren
 
     @Override
     public void bind(@NonNull PlainPost post) {
+        if (post instanceof SinglePost) {
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/ui/adapters/holders/ViewHolderPost.java]: Do you need it now?

File: app/src/main/java/ly/loud/loudly/ui/feed/FeedFragment.java [hunk 3]
```diff
@@ -143,6 +150,18 @@ public void onLikesClick(@NonNull PlainPost post) {
 
     @Override
     public void onDeleteClick(@NonNull PlainPost post) {
-        presenter.deletePost(post);
+        if (post instanceof LoudlyPost) {
+            showDefaultSnackBar(R.string.message_start_post_deletion);
+            presenter.deletePost((LoudlyPost) post);
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/ui/feed/FeedFragment.java]: It's better to show confirmation about deletion

File: app/src/main/java/ly/loud/loudly/ui/new_post/NewPostFragment.java [hunk 4]
```diff
@@ -100,6 +110,23 @@ public void showGalleryError() {
 
     }
 
+    /**
+     * Triggered by presenter when loudlyPost uploaded to one more network.
+     */
+    @Override
+    public void onPostUploadingProgress(@NonNull LoudlyPost loudlyPost) {
+        // Just notify activity about progress
+        ((NewPostFragmentInteractions) getActivity()).onPostUploadProgress(loudlyPost);
```
Comment (danilkolikov) [inline app/src/main/java/ly/loud/loudly/ui/new_post/NewPostFragment.java]: Here you may got ClassCastException, I we should use setter of callback (as in other fragments)

Comment (ZeRoGerc) [inline app/src/main/java/ly/loud/loudly/ui/new_post/NewPostFragment.java]: fixed
@see onAttach method

**Comment by danilkolikov** [comment]
ok

**Pull request merged**